kedro/framework/context/context.py

```diff
@@ -154,7 +155,9 @@ def _update_nested_dict(old_dict: Dict[Any, Any], new_dict: Dict[Any, Any]) -> N
         if key not in old_dict:
             old_dict[key] = value
         else:
-            if isinstance(old_dict[key], dict) and isinstance(value, dict):
+            if isinstance(old_dict[key], (dict, DictConfig)) and isinstance(
```

Comment:
Maybe a better way would be to check against the superclass of all dict-like objects to remove the strict dependency on `omegaconf` and make this future-proof.

```python
from collections.abc import Mapping
from omegaconf import DictConfig

assert isinstance({}, Mapping) 
assert isinstance(DictConfig({1:1}), Mapping)
```

I would also double check for other occurrences of `isinstance(x, dict)` in the code

Reply:
This is creating trouble with `mypy` in `_update_nested_dict`. 😅